common/presenters/move-to-message-banner-component.js

```diff
@@ -0,0 +1,164 @@
+const { isEmpty } = require('lodash')
+
+const i18n = require('../../config/i18n')
+const filters = require('../../config/nunjucks/filters')
+const componentService = require('../services/component')
+
+const frameworkToTaskListComponent = require('./framework-to-task-list-component')
+
+function _returnComponent(title, content) {
+  return {
+    allowDismiss: false,
+    classes: 'app-message--instruction govuk-!-padding-right-0',
+    title: {
+      text: title,
+    },
+    content: {
+      html: content,
+    },
+  }
+}
+
+function moveToMessageBannerComponent({ move, moveUrl, canAccess }) {
+  const { profile, status } = move
+  const { id: profileId, person_escort_record: personEscortRecord } =
+    profile || {}
+
+  // TODO: Add logic to determine which assessment to show, when
+  const context = 'person-escort-record'
+  const type = 'person-escort-record'
+  const baseUrl = `${moveUrl}/person-escort-record`
+
+  if (['proposed'].includes(status) || !profileId) {
+    return undefined
+  }
+
+  let title
+  let content
+
+  // No PER exists
+  if (['requested', 'booked'].includes(status) && isEmpty(personEscortRecord)) {
+    title = i18n.t('messages::assessment.pending.heading', {
+      context,
+      type,
+    })
+    content = `<p>${i18n.t('messages::assessment.pending.content', {
+      context,
+      type,
+    })}</p>`
+
+    if (canAccess('person_escort_record:create')) {
+      content += `
+        ${componentService.getComponent('govukButton', {
+          href: `${baseUrl}/new`,
+          text: i18n.t('actions::start_assessment', {
+            context,
+            type,
+          }),
+        })}
+      `
+    }
+
+    return _returnComponent(title, content)
```

Comment:
Perhaps `_returnComponent` could return undefined if there's no content passed - it would remove the need for multiple returns here and result in just one exit point for the larger function whose responsibility then becomes to just figure out what title and content are rather than also whether the component exists or not.

Reply:
So, I originally had something similar where these if blocks weren't returning anything and there was a single call at the end.

However, that meant in some circumstances two blocks were being entered before the return.

I was trying to take more of a [return early pattern](https://medium.com/swlh/return-early-pattern-3d18a41bba8) instead of requiring the use of `else` or more statements within each `if` block.